tools/prepare_software_list.py

```diff
@@ -0,0 +1,290 @@
+DOAP_NAME = f'.//{{{DOAP_NS}}}name'
+DOAP_SHORTDESC = f'.//{{{DOAP_NS}}}shortdesc'
+DOAP_OS = f'.//{{{DOAP_NS}}}os'
+DOAP_LOGO = f'.//{{{SCHEMA_NS}}}logo'
+
+PLATFORMS: list[str] = [
+    'Android',
+    'iOS',
+    'Windows',
+    'macOS',
+    'Linux',
+]
+
+ENTRY_LIFETIME = timedelta(days=365)
+
+
+def initialize_directory(path: Path) -> None:
+    '''
+    Remove path (if it exists) and containing files, then recreate path
+    '''
+    if path.exists() and path.is_dir():
+        shutil.rmtree(path)
+        os.mkdir(path)
+    else:
+        os.mkdir(path)
+
+
+def download_file(url: str, path: Path) -> bool:
+    '''
+    Downloads file from url and stores it in /downloads/path
+    returns success
+    '''
+    file_request = requests.get(url)
+    if not 200 >= file_request.status_code < 400:
+        print('Error while trying to download from', url)
+        return False
+
+    with open(DOWNLOAD_PATH / path, 'wb') as file_data:
+        file_data.write(file_request.content)
+    return True
+
+
+def check_renewal(name: str, renewed: Optional[str]) -> bool:
+    '''
+    Check if 'last_renewed' entry is not older than 365 days
+    '''
+    if renewed is None:
+        return False
+    try:
+        renewed = datetime.strptime(renewed, "%Y-%m-%dT%H:%M:%S")
+    except ValueError:
+        print('Failed to parse timestamp for', name)
+        return False
+    now = datetime.utcnow()
+    if now - renewed > ENTRY_LIFETIME:
+        print('Client entry expired for', name)
+        return False
+    return True
+
+
+def parse_doap_infos(doap_file: str
+                     ) -> Optional[dict[str, Union[str, list[str], None]]]:
+    '''Parse DOAP file and return infos'''
+    try:
+        doap = ET.parse(
+            DOWNLOAD_PATH / f'doap_files/{doap_file}.doap')
+    except FileNotFoundError:
+        return None
+
+    info: dict[str, Union[str, list[str], None]] = {}
+
+    info['name'] = None
+    doap_name = doap.find(DOAP_NAME)
+    if doap_name is not None:
+        info['name'] = doap_name.text
+
+    info['shortdesc'] = None
+    doap_shortdesc = doap.find(DOAP_SHORTDESC)
+    if doap_shortdesc is not None:
+        info['shortdesc'] = doap_shortdesc.text
+
+    info['os'] = []
+    for entry in doap.findall(DOAP_OS):
+        info['os'].append(entry.text)
+    if not info['os']:
+        info['os'] = ['Other']
+
+    info['logo'] = None
+    doap_logo = doap.find(DOAP_LOGO)
+    if doap_logo is not None:
+        info['logo'] = doap_logo.attrib[RDF_RESOURCE]
+
+    return info
+
+
+def check_file_size(file_path: Path) -> None:
+    '''
+    Check if file size is greater than 300 KiB and if so, resize image
+    '''
+    try:
+        file_size = os.path.getsize(file_path)
+    except OSError as error:
+        print('An error occurred while trying to open logo:', error)
+    if file_size > 300000:
+        try:
+            with Image.open(file_path) as img:
```

Comment:
I'm not sure I'm happy with PIL processing untrusted data, but I guess that is ok if we ensure that pillow and the backing libraries are up-to-date.

Reply:
At least `Pillow` is installed via `pip install --upgrade`, and I didn't pin any version in `requirements.txt`.

Reply:
Website builds are now available via Github Actions. This should relax the situation.